src/core_codemods/file_resource_leak.py

```diff
@@ -21,70 +26,75 @@
     Metadata,
     Reference,
     ReviewGuidance,
-    SimpleCodemod,
 )
+from core_codemods.api.core_codemod import CoreCodemod
 
 
-class FileResourceLeak(SimpleCodemod):
-    metadata = Metadata(
-        name="fix-file-resource-leak",
-        summary="Automatically Close Resources",
-        review_guidance=ReviewGuidance.MERGE_WITHOUT_REVIEW,
-        references=[
-            Reference(url="https://cwe.mitre.org/data/definitions/772.html"),
-            Reference(url="https://cwe.mitre.org/data/definitions/404.html"),
-        ],
-    )
+class FileResourceLeakTransformer(LibcstResultTransformer):
     change_description = "Wrapped opened resource in a with statement."
 
     METADATA_DEPENDENCIES = (
-        PositionProvider,
         ScopeProvider,
         ParentNodeProvider,
     )
 
     def __init__(
         self,
         context: CodemodContext,
+        results: list[Result],
         file_context: FileContext,
-        *codemod_args,
+        *codemod_args,  # pylint: disable=unused-argument
```

Comment:
Small thing but I think the idiomatic way to suppress this is with
```python
del *codemod_args
```

Although possibly this argument is no longer needed?